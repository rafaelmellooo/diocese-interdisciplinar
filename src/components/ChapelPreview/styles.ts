import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    card: {
        height: 220,
        backgroundColor: '#ffffff',
        marginHorizontal: 10,
        borderTopLeftRadius: 5,
        borderTopRightRadius: 5,
        padding: 10,
        alignItems: 'center',
    },
    scheduleButton: {
        backgroundColor: '#1386F2',
        paddingVertical: 10,
        paddingHorizontal: 20,
        flexDirection: 'row',
        alignItems: 'center',
        borderRadius: 5,
    },
    scheduleButtonText: {
        color: '#ffffff',
        fontSize: 16,
        marginLeft: 6
    }
});