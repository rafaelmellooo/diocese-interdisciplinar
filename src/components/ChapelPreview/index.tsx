import React from 'react';
import { Text, View, TouchableOpacity, useWindowDimensions } from 'react-native';
import { Ionicons } from '@expo/vector-icons'
import { useNavigation, useTheme } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';

import { styles } from "./styles";
import { RootStackParamList } from '../../navigation';

type ChapelPreviewProps = {
    name: string;
    info: string;
    distance: string;
    address: string;
}

export default function ChapelPreview(props: ChapelPreviewProps) {
    const navigation = useNavigation<StackNavigationProp<RootStackParamList, 'Home'>>();

    const { dark, colors } = useTheme();

    const dimensions = useWindowDimensions();

    const handleReminderButtonPress = () => {
        navigation.navigate('NewReminder', {
            name: props.name,
            info: props.info,
            distance: props.distance
        });
    }

    return (
        <View style={[
            styles.card,
            {
                width: dimensions.width * 0.8,
                backgroundColor: colors.card
            }
        ]}>
            <View style={{
                flexDirection: 'row',
                alignItems: 'flex-start',
            }}>
                <View>
                    <Text style={[styles.title, {
                        color: dark ? '#1386F2' : '#0D2744'
                    }]}>
                        {props.name}
                    </Text>

                    <Text style={[{
                        color: colors.text
                    }]}>
                        {props.info}
                    </Text>

                    <Text style={[{
                        color: colors.text
                    }]}>
                        {props.address}
                    </Text>
                </View>

                <Text style={[styles.distance, {
                    borderColor: colors.border,
                }]}>
                    {props.distance}
                </Text>
            </View>

            <TouchableOpacity style={[styles.scheduleButton, {
                backgroundColor: colors.primary
            }]} onPress={() => handleReminderButtonPress()}>
                <Ionicons name='notifications' color={dark ? '#0A1D33' : '#F5F3F3'} size={16} />

                <Text
                    style={[styles.scheduleButtonText, {
                        color: dark ? '#0A1D33' : '#F5F3F3'
                    }]}
                >
                    Criar Lembrete
                </Text>
            </TouchableOpacity>
        </View>
    );
}