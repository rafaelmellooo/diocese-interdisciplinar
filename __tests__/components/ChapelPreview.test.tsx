import React from "react";
import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import ChapelPreview from "../../src/components/ChapelPreview";

const mockProps = {
  name: "Chapel Name",
  info: "Chapel Info",
  distance: "5 km",
  address: "Chapel Address",
};

it("renders correctly", () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <ChapelPreview {...mockProps} />
      </NavigationContainer>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
