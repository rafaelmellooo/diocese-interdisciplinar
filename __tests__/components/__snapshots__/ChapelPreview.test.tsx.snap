// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "height": 220,
        "marginHorizontal": 10,
        "padding": 10,
      },
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "width": 600,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 10,
            },
            {
              "color": "#0D2744",
            },
          ]
        }
      >
        Chapel Name
      </Text>
      <Text
        style={
          [
            {
              "color": "rgb(28, 28, 30)",
            },
          ]
        }
      >
        Chapel Info
      </Text>
      <Text
        style={
          [
            {
              "color": "rgb(28, 28, 30)",
            },
          ]
        }
      >
        Chapel Address
      </Text>
    </View>
    <Text
      style={
        [
          {
            "borderRadius": 20,
            "borderWidth": 2,
            "color": "#1386F2",
            "fontSize": 18,
            "paddingHorizontal": 8,
            "paddingVertical": 4,
          },
          {
            "borderColor": "rgb(216, 216, 216)",
          },
        ]
      }
    >
      5 km
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 122, 255)",
        "borderRadius": 5,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text />
    <Text
      style={
        [
          {
            "fontSize": 16,
            "marginLeft": 6,
          },
          {
            "color": "#F5F3F3",
          },
        ]
      }
    >
      Criar Lembrete
    </Text>
  </View>
</View>
`;
